app/src/main/java/com/example/healthinspector/CreateRecommendations.java

```diff
@@ -0,0 +1,170 @@
+
+                                StringBuilder keywordsBuilder = new StringBuilder();
+                                JSONArray productKeywords = recommendedItemJson.getJSONArray(KEYWORDS);
+                                for (int x = 0; x < Integer.min(KEYWORD_LIMIT,productKeywords.length()); x++){
+                                    keywordsBuilder.append(productKeywords.getString(x)).append("%20");
+                                }
+                                keywords = keywordsBuilder.toString();
+
+                                imageUrl = recommendedItemJson.getString(Constants.PRODUCT_IMAGE);
+                                if(imageUrl == null){
+                                    imageUrl = "";
+                                }
+                                if(recommendedItemJson.has(Constants.NUTRIENT_LEVELS)){
+                                    JSONObject nutrientLevelsJSON = recommendedItemJson.getJSONObject(Constants.NUTRIENT_LEVELS);
+                                    Iterator iterator =  nutrientLevelsJSON.keys();
+                                    while (iterator.hasNext()){
+                                        String key = iterator.next().toString();
+                                        nutrientLevels.add(key + ": " + nutrientLevelsJSON.getString(key));
+                                    }
+                                }
+                            } catch (JSONException e) {
+                                e.printStackTrace();
+                            }
+                            recommendedProducts.add(new RecommendedProduct(keywords, brand, productName, imageUrl, nutrientLevels));
+                        }
+                        ProgressBar progressBar;
+                        if(fragmentSwitch.equals(FragmentSwitch.RECOMMENDATIONS)){
+                          progressBar = v.findViewById(R.id.progressBar);
+                        }
+                        else{
+                          progressBar = v.findViewById(R.id.homeProgressBar);
+                        }
+                        progressBar.setVisibility(View.GONE);
+                        try {
+                            loadRecommendationsIntoView(recommendedProducts, fragmentSwitch, v, context, scannedProduct);
+                        } catch (ParseException e) {
+                            Log.e(TAG, "ParseException trying to get recommended products " + e);
+                            Toast.makeText(context, context.getString(R.string.error_creating_recommendations), Toast.LENGTH_SHORT).show();
+                            e.printStackTrace();
+                        } catch (JSONException e) {
+                            Log.e(TAG, "JSONException trying to get recommended products " + e);
+                            Toast.makeText(context, context.getString(R.string.error_creating_recommendations), Toast.LENGTH_SHORT).show();
+                            e.printStackTrace();
```

Comment:
these are identical, possibly catch both exceptions in one block?

Reply:
https://www.geeksforgeeks.org/multicatch-in-java/#:~:text=Starting%20from%20Java%207.0%2C%20it,code%20duplication%20and%20increases%20efficiency.